core/src/main/java/com/linecorp/armeria/common/RedirectService.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.linecorp.armeria.server.AbstractHttpService;
+import com.linecorp.armeria.server.PathMapping;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * An {@link AbstractHttpService} that implements a Http Redirect.
+ */
+
+public class RedirectService extends AbstractHttpService {
+
+    private final PathMapping pm;
+    private final HttpStatus httpStatus;
+    private final Function<? super ServiceRequestContext, String> callback;
+
+    /**
+     * RedirectService.
+     */
+    public RedirectService(String realPath) {
+        this(HttpStatus.TEMPORARY_REDIRECT, realPath);
+    }
+
+    /**
+     * RedirectService.
+     */
+    public RedirectService(Function<? super ServiceRequestContext, String> callback) {
+        this(HttpStatus.TEMPORARY_REDIRECT, callback);
+    }
+
+    /**
+     * RedirectService.
+     */
+    public RedirectService(HttpStatus redirectStatus, String realPath) {
+        requireNonNull(redirectStatus, "redirectStatus");
+        requireNonNull(realPath, "realPath");
+        if (redirectStatus.compareTo(HttpStatus.MULTIPLE_CHOICES) < 0 ||
+                redirectStatus.compareTo(HttpStatus.TEMPORARY_REDIRECT) > 0) {
+            throw new IllegalArgumentException(redirectStatus + " is not a redirect status");
```

Comment:
How about: `"redirectStatus: " + redirectStatus + " (expected: 300 .. 307)"` (for consistency)